# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/lvm2/linux-2.6.10-fs_h.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur LVM2.2.00.25/lib/device/dev-io.c src.lvm2.1104330759.23110.469340415/LVM2.2.00.25/lib/device/dev-io.c
--- LVM2.2.00.25/lib/device/dev-io.c	2004-09-14 22:23:23.000000000 +0000
+++ src.lvm2.1104330759.23110.469340415/LVM2.2.00.25/lib/device/dev-io.c	2004-12-29 14:42:55.934962456 +0000
@@ -31,7 +31,11 @@
 #  define u64 uint64_t		/* Missing without __KERNEL__ */
 #  undef WNOHANG		/* Avoid redefinition */
 #  undef WUNTRACED		/* Avoid redefinition */
-#  include <linux/fs.h>		/* For block ioctl definitions */
+/* copied from: <linux/fs.h> */	/* For block ioctl definitions */
+#  define BLKBSZGET  _IOR(0x12,112,size_t)
+#  define BLKSSZGET  _IO(0x12,104)/* get block device sector size */
+#  define BLKFLSBUF  _IO(0x12,97)	/* flush buffer cache */
+
 #  define BLKSIZE_SHIFT SECTOR_SHIFT
 #  ifndef BLKGETSIZE64		/* fs.h out-of-date */
 #    define BLKGETSIZE64 _IOR(0x12, 114, size_t)
